src/InsertCapablePdoTrait.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use ArrayAccess;
+use Dhii\Util\String\StringableInterface;
+use OutOfRangeException;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use InvalidArgumentException;
+use PDOStatement;
+use Psr\Container\ContainerInterface;
+use stdClass;
+use Traversable;
+
+/**
+ * Common functionality for objects that can insert records into a database using PDO.
+ *
+ * @since [*next-version*]
+ */
+trait InsertCapablePdoTrait
+{
+    /**
+     * Executes an INSERT SQL query, inserting records into the database.
+     *
+     * @since [*next-version*]
+     *
+     * @param array[]|ArrayAccess[]|stdClass[]|ContainerInterface[]|Traversable $records A list of records to insert.
+     *
+     * @throws ContainerExceptionInterface If an error occurred while reading from a record's container.
+     *
+     * @return PDOStatement The executed PDO statement.
+     */
+    protected function _insert($records)
+    {
+        $processedRecords = $this->_preProcessRecords($records, $valueHashMap);
```

Comment:
Explain why this is necessary. AFAIU, it turns each key-value mapping in the record into key-valhash. But from here, this is not visible.

Reply:
Perhaps rename and redoc the method? I remember struggling to name this method, and so just called the procedure "pre-processing". If so, any suggestions?